scipy/optimize/__minpack.h

```diff
@@ -88,13 +88,22 @@ int jac_multipack_calling_function(int *n, double *x, double *fvec, double *fjac
   */
 
   PyArrayObject *result_array;
+  npy_intp fvec_sz = 0;
+  int res = 0;
 
   if (*iflag == 1) {
     result_array = (PyArrayObject *)call_python_function(multipack_python_function, *n, x, multipack_extra_arguments, 1, minpack_error);
     if (result_array == NULL) {
       *iflag = -1;
       return -1;
     }
+    fvec_sz = PyArray_SIZE(result_array);
+    if((int)fvec_sz != *n){
```

Comment:
Best to add a `npy_intp out_size` argument to `call_python_function`, and put the check there, so that you don't need to add new error handling branches everywhere.

Since `call_python_function` is used also for the first call, you can make it so that the check is skipped for `out_size == -1`.